app/WebParsing/ReqParser.hs

```diff
@@ -26,6 +26,32 @@ completionPrefix = Parsec.choice (map (Parsec.try . Parsec.string) [
     ])
     >> Parsec.spaces
 
+programPrefix :: Parser ()
+programPrefix = Parsec.choice (map caseInsensitiveStr [
+    "admission to",
+    "enrolment in the ",
+    "enrolment in an ",
+    "enrolment in a ", -- trailing space to prevent matching "enrolment in A..."
```

Comment:
This is an excellent point. Really, though, *each* of the phrases in this category should end with a space (even `admission to`). So rather than add a space manually to these phrases, you should parse a single space character *after* this part.

This could look like an extra `>>` before the `Parsec.spaces`, or you can use [`skipMany1`](https://hackage.haskell.org/package/parsec-3.1.11/docs/Text-Parsec.html#v:skipMany1) combined with `isSpace` to check for 1 or more spaces instead of 0 or more.